docs/proposals/20220725-managed-kubernetes.md

```diff
@@ -0,0 +1,493 @@
+
+### Goals
+
+- Provide a recommendation of a consistent approach for representing managed Kubernetes services in CAPI for new implementations.
+- Reach a consensus on how ClusterClass should be supported for managed Kubernetes
+- As a result of the recommendations of this proposal we should update the [Provider Implementers](https://cluster-api.sigs.k8s.io/developer/providers/implementers.html) documentation to aid with future provider implementations.
+
+### Non-Goals/Future Work
+
+- Require the existing implementations in CAPA & Cluster API Provider Azure (CAPZ) to converge on the chosen approach.
+  - It would be ideal for there to be consistency between providers when it comes to representing managed Kubernetes services. However, it's unrealistic to ask providers to refactor their implementations.
+  - If providers such as CAPA & CAPZ would like to follow the guidance then discussions should be facilitated.
+- Provide advice in this proposal on how to refactor the existing implementations of managed Kubernetes in CAPA & CAPZ.
+- Propose a new architecture or API changes to CAPI for managed Kubernetes
+- Be a concrete design for the GKE implementation in Cluster API Provider GCP (CAPG).
+  - A separate proposal will be created for CAPG based on the recommendations of this proposal.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a cluster operator,
+I want to use Cluster API to provision and manage the lifecycle of a control plane that utilizes my service provider's managed Kubernetes control plane (i.e. EKS, AKS, GKE),
+So that I don’t have to worry about the management/provisioning of control plane nodes, and so I can take advantage of any value add services offered by the service provider.
+
+#### Story 2
+
+As a cluster operator,
+I want to be able to provision both “unmanaged” and “managed” Kubernetes clusters from the same management cluster,
+So that I can support different requirements & use cases as needed whilst using a single operating model.
+
+#### Story 3
+
+As a Cluster API provider developer,
+I want guidance on how to incorporate a managed Kubernetes service into my provider,
+So that its usage is compatible with Cluster API architecture/features
+And its usage is consistent with other providers.
+
+#### Story 4
+
+As a Cluster API provider developer,
+I want to enable the ClusterClass feature for a managed Kubernetes service,
+So that users can take advantage of an improved UX with ClusterClass-based clusters.
+
+#### Story 5
+
+As a cluster operator,
+I want to use Cluster API to provision and manage the lifecycle of worker nodes that utilizes my cloud providers’ managed instances (if they support them),
+So that I don't have to worry about the management of these instances.
+
+#### Story 6
+
+As a Service provider,
+I want to be able to offer “Managed Kubernetes” clusters via a CAPI provider which is consistent with other implementations,
+So that users intuitively know how to create clusters with my service.
```

Comment:
I'd be more specific about "users" here, e.g:
As a Service provider, I want to be able to offer “Managed Kubernetes” powered by CAPI so I can eliminate the responsibility of owning and SREing the Control Plane from the Cluster consumer and cluster admin (or cluster operator).

Reply:
updated